haskell/private/haskell_impl.bzl

```diff
@@ -179,14 +217,41 @@ def _haskell_binary_common_impl(ctx, is_test):
         bin_info = bin_info,
     )
 
+    executable = binary
+
+    if inspect_coverage:
+        binary_path = ctx.workspace_name + "/" + binary.short_path
+        hpc_path = ctx.workspace_name + "/" + hs.toolchain.tools.hpc.short_path
+        tix_file_path = hs.label.name + ".tix"
+        mix_file_paths = ""
```

Comment:
why this is not a list?

Reply:
I did it this way since it needs to be inserted into the bash script this way. I could add a couple steps in here and make it nicer, for sure.

Reply:
I gave your suggestions a shot and I actually don't feel bad about the current implementation. If I used a list comprehension, I would still have to iterate through again and concatenate the strings with a for loop, since Starlark's lists have no .join(...) method. Remember, this is to get the filenames to be of form "a.txt b.txt c.txt" for insertion into the bash template.

If I'm missing something here, please let me know. I'll leave this unresolved for the moment.

Reply:
@mboes trivia about python (the CPython implementation, I don't have any idea about skylark), but string concatenation with `a + b` is `O(len(b))` with potentially 0 allocation (depending on the overallocated space in `a`). So string building with concatenation is basically `O(n)`.

All of that thanks to a check on the reference counter.

Reply:
https://docs.bazel.build/versions/master/skylark/lib/string.html#join for a join method.

Reply:
With Philip's advice below, I've used a list comprehension along with using shell.array_literal.